packages/rpc-provider/src/substrate-connect/index.ts

```diff
@@ -154,9 +154,17 @@ export class ScProvider implements ProviderInterface {
       callback?.(decodedResponse);
     };
 
-    const addChain = this.#wellKnownChains.has(this.#spec as ScType.WellKnownChain)
-      ? client.addWellKnownChain
-      : client.addChain;
+    const addChain = this.#sharedSandbox
+      ? (async (...args) => {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const source = this.#sharedSandbox!;
```

Comment:
`eslint-disable` because it's checked on line [157](https://github.com/polkadot-js/api/pull/5776/files#diff-66ebd813ed4a22dc4419b70f5b5771ac5f597162674c041aad2ecec74717fd0dR157)